cmd/pipeline/deploy.go

```diff
@@ -331,3 +297,90 @@ func CheckAllResourcesRunning(name string, resourceStatus map[string]string) (bo
 	}
 	return allRunning, nil
 }
+
+func (f deployFlags) toOptions() *DeployOptions {
+	file := f.file
+	if f.filename != "" {
+		oktetoLog.Warning("the 'filename' flag is deprecated and will be removed in a future version. Please consider using 'file' flag")
+		if file == "" {
+			file = f.filename
+		} else {
+			oktetoLog.Warning("flags 'filename' and 'file' can not be used at the same time. 'file' flag will take precedence")
+		}
+	}
+	return &DeployOptions{
+		Branch:       f.branch,
+		Repository:   f.repository,
+		Name:         f.name,
+		Namespace:    f.namespace,
+		Wait:         f.wait,
+		SkipIfExists: f.skipIfExists,
+		Timeout:      f.timeout,
+		File:         file,
+		Variables:    f.variables,
+	}
+}
+
+func (o *DeployOptions) setDefaults() error {
+	cwd, err := os.Getwd()
+	if err != nil {
+		return fmt.Errorf("failed to get the current working directory: %w", err)
+	}
+
+	if o.Repository == "" {
+		oktetoLog.Info("inferring git repository URL")
+
+		o.Repository, err = model.GetRepositoryURL(cwd)
+		if err != nil {
+			return fmt.Errorf("could not get repository url: %w", err)
+		}
+	}
+
+	if o.Name == "" {
+		o.Name = getPipelineName(o.Repository)
+	}
+
+	currentRepo, err := model.GetRepositoryURL(cwd)
+	if err != nil {
+		oktetoLog.Debug("cwd does not have .git folder")
+	}
+
+	if o.Branch == "" && okteto.AreSameRepository(o.Repository, currentRepo) {
+
+		oktetoLog.Info("inferring git repository branch")
+		b, err := utils.GetBranch(cwd)
+
+		if err != nil {
+			return err
+		}
+
+		o.Branch = b
+	}
+
+	if o.Namespace == "" {
+		o.Namespace = okteto.Context().Namespace
+	}
+	return nil
+}
+
+func (o *DeployOptions) toPipelineDeployOptions() (types.PipelineDeployOptions, error) {
```

Comment:
Can we make this more specific to reference that is for the client? I mean, I think it might be confusing to have a function called `toOptions` and another called `toPipelineDeployOptions` in a struct called `DeployOptions` in a `pipeline` package. What is the difference between both?

Maybe `toPipelineDeployClientOptions`?